src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -571,77 +659,127 @@ public static bool TryParse(string version, out SemanticVersion result)
 
         private static bool TryParseVersion(string version, ref VersionResult result)
         {
+            if (version.EndsWith('-') || version.EndsWith('+') || version.EndsWith('.'))
+            {
+                result.SetFailure(ParseFailureKind.FormatException);
+                return false;
+            }
+
+            string versionSansLabel = null;
+            var major=0;
+            var minor=0;
+            var patch=0;
+            string preLabel = null;
+            string buildLabel = null;
+
             var dashIndex = version.IndexOf('-');
+            var plusIndex = version.IndexOf('+');
```

Comment:
This parse code would be easier to read if it included comments with examples of expected version string.

Reply:
Comment added.